kernel/h1_syscalls/src/spi_device.rs

```diff
@@ -0,0 +1,142 @@
+use core::cell::Cell;
+use h1::hil::spi_device::{SpiDevice, SpiDeviceClient};
+use kernel::{AppId, Callback, Driver, Grant, ReturnCode, Shared, AppSlice};
+
+pub const DRIVER_NUM: usize = 0x40030;
+
+#[derive(Default)]
+pub struct AppData {
+    tx_buffer: Option<AppSlice<Shared, u8>>,
+    rx_buffer: Option<AppSlice<Shared, u8>>,
+    data_received_callback: Option<Callback>,
+}
+
+pub struct SpiDeviceSyscall<'a> {
+    device: &'a dyn SpiDevice,
+    apps: Grant<AppData>,
+    current_user: Cell<Option<AppId>>,
+}
+
+impl<'a> SpiDeviceSyscall<'a> {
+    pub fn new(device: &'a dyn SpiDevice,
+               container: Grant<AppData>) -> SpiDeviceSyscall<'a> {
+        SpiDeviceSyscall {
+            device: device,
+            apps: container,
+            current_user: Cell::new(None),
+        }
+    }
+
+    fn send_data(&self, caller_id: AppId, clear_busy: bool) -> ReturnCode {
+        self.apps.enter(caller_id, |app_data, _| {
+            if let Some(ref tx_buffer) = app_data.tx_buffer {
+                debug!("send_data: clear_busy={:?}", clear_busy);
```

Comment:
Is this `debug!` call left in intentionally? There are a few others in this file as well.

Reply:
I was on the fence of whether to leave them in for the time being. But I think you're right, it's cleaner to keep them, but commented out. I'll revisit once the thing has stabilized on removing them completely.